src/main/java/appeng/client/render/cablebus/FacadeBuilder.java

```diff
@@ -411,8 +405,56 @@ private static AEAxisAlignedBB getCutOutBox( AxisAlignedBB facadeBox, List<AxisA
 	}
 
 	/**
-	 * Determines if any of the part's bounding boxes intersects with the outside 2 voxel wide layer.
-	 * If so, we should use thinner facades (1 voxel deep).
+	 * Generates the box segments around the specified hole. If the specified hole is null, a Singleton of the Facade box is returned.
+	 *
+	 * @param fb The Facade's box.
+	 * @param hole The hole to 'cut'.
+	 * @param axis The axis the facade is on.
+	 *
+	 * @return The box segments.
+	 */
+	private static List<AxisAlignedBB> getBoxes( AxisAlignedBB fb, AEAxisAlignedBB hole, Axis axis )
+	{
+		if( hole == null )
+		{
+			return Collections.singletonList( fb );
+		}
+		List<AxisAlignedBB> boxes = new ArrayList<>();
+		switch( axis )
+		{
+			case Y:
+				boxes.add( new AxisAlignedBB( fb.minX, fb.minY, fb.minZ, hole.minX, fb.maxY, fb.maxZ ) );
+				boxes.add( new AxisAlignedBB( hole.maxX, fb.minY, fb.minZ, fb.maxX, fb.maxY, fb.maxZ ) );
+
+				boxes.add( new AxisAlignedBB( hole.minX, fb.minY, fb.minZ, hole.maxX, fb.maxY, hole.minZ ) );
+				boxes.add( new AxisAlignedBB( hole.minX, fb.minY, hole.maxZ, hole.maxX, fb.maxY, fb.maxZ ) );
+
+				break;
+			case Z:
+				boxes.add( new AxisAlignedBB( fb.minX, fb.minY, fb.minZ, fb.maxX, hole.minY, fb.maxZ ) );
+				boxes.add( new AxisAlignedBB( fb.minX, hole.maxY, fb.minZ, fb.maxX, fb.maxY, fb.maxZ ) );
+
+				boxes.add( new AxisAlignedBB( fb.minX, hole.minY, fb.minZ, hole.minX, hole.maxY, fb.maxZ ) );
+				boxes.add( new AxisAlignedBB( hole.maxX, hole.minY, fb.minZ, fb.maxX, hole.maxY, fb.maxZ ) );
+
+				break;
+			case X:
+				boxes.add( new AxisAlignedBB( fb.minX, fb.minY, fb.minZ, fb.maxX, hole.minY, fb.maxZ ) );
+				boxes.add( new AxisAlignedBB( fb.minX, hole.maxY, fb.minZ, fb.maxX, fb.maxY, fb.maxZ ) );
+
+				boxes.add( new AxisAlignedBB( fb.minX, hole.minY, fb.minZ, fb.maxX, hole.maxY, hole.minZ ) );
+				boxes.add( new AxisAlignedBB( fb.minX, hole.minY, hole.maxZ, fb.maxX, hole.maxY, fb.maxZ ) );
+				break;
+			default:
+				//should never happen.
+				throw new RuntimeException( "switch falloff. " + String.valueOf( axis ) );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00112)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Directly append the argument of String.valueOf(). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1153)